project/src/main/kotlin/org/ionproject/core/calendar/CalendarController.kt

```diff
@@ -0,0 +1,64 @@
+package org.ionproject.core.calendar
+
+import org.ionproject.core.common.Media
+import org.ionproject.core.common.Uri
+import org.springframework.http.ResponseEntity
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.RestController
+
+@RestController
+class CalendarController(private val repository: CalendarRepo) {
+
+    @GetMapping(Uri.calendarByClass, produces = [Media.CALENDAR, Media.SIREN_TYPE])
+    fun fromClass(
+        @PathVariable cid: Int,
+        @PathVariable calterm: String
+    ): ResponseEntity<Any> {
+        val calendar = repository.getClassCalendar(cid, calterm)
+        return if (calendar != null)
+            ResponseEntity.ok(calendar)
+        else
+            ResponseEntity.notFound().build()
+    }
+
+    @GetMapping(Uri.calendarByClassSection, produces = [Media.CALENDAR, Media.SIREN_TYPE])
+    fun fromClassSection(
+        @PathVariable sid: String,
+        @PathVariable calterm: String,
+        @PathVariable cid: Int
+    ): ResponseEntity<Any> {
+        val calendar = repository.getClassSectionCalendar(cid, calterm, sid)
+        return if (calendar != null)
+            ResponseEntity.ok(calendar)
+        else
+            ResponseEntity.notFound().build()
+    }
+
+    @GetMapping(Uri.componentByClassCalendar, produces = [Media.CALENDAR, Media.SIREN_TYPE])
+    fun fromClassCalendar(
+        @PathVariable cid: Int,
+        @PathVariable calterm: String,
+        @PathVariable cmpid: String
```

Comment:
Error, it should name "cmpid" should match the same name as in the template path. (Component) "${version}/courses/{cid}/classes/{calterm}/calendar/{component}"